sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SessionsMessagePump.java

```diff
@@ -284,6 +287,16 @@ private void throwIfTerminatedOrInitialized() {
         }
     }
 
+    private void logCPUResourcesConcurrencyMismatch() {
+        final int cores = Runtime.getRuntime().availableProcessors();
+        final int poolSize = DEFAULT_BOUNDED_ELASTIC_SIZE;
+        final int concurrency = maxConcurrentSessions * concurrencyPerSession;
+        if (concurrencyPerSession > poolSize || concurrency > CONCURRENCY_PER_CORE * cores) {
+            final String message = concurrency + " (ConcurrentSessions=" + maxConcurrentSessions + ", ConcurrencyPerSession=" + concurrencyPerSession + ")";
+            logger.atInfo().log(CORES_VS_CONCURRENCY_MESSAGE, poolSize, cores, message);
```

Comment:
wdyt about warning?

Reply:
right, this make sense to be in WARN level, I'll update.